pandas/core/internals/__init__.py

```diff
@@ -2785,6 +2787,17 @@ def set(self, locs, values, check=False):
 
         self.values[locs] = values
 
+    def eval(self, func, other, try_cast=False, **kwargs):
+        block = super().eval(func, other, try_cast=try_cast, **kwargs)[0]
+        if try_cast:
+            if isinstance(other,
+                          (tslibs.Timestamp, np.datetime64, datetime, date)):
```

Comment:
Is there an ABCTimestamp to check? At least 3 times use similar `isinstance()`.

Reply:
> Is there an ABCTimestamp to check?

No, but Timestamp subclasses datetime, so could actually be removed from this check.